src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4901,6 +4901,11 @@ private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> v
                     else
                     {
                         var targetType = variable.Type;
+                        if (variable.Expression.ExpressionSymbol is SourceLocalSymbol local && local.IsVar && rightPart is BoundFieldAccess rightField)
```

Comment:
>local.IsVar [](start = 95, length = 11)

Will this re-infer a `var` local, even if the local was declared earlier? For instance `(x2, y2) = u` below:
```C#
    static void F(object? x1, object y1)
    {
        var t = (x1, y1);
        var (x2, y2) = t;
        if (x1 == null) return;
        var u = (x1, y1);
        (x2, y2) = u;
    }
```

Reply:
Added a test to cover this.